Move tree utility functions into utils module

huddlej opened the pull request:
Moves `all_parents` function out of the clades module and into the utils module,
to make this generic utility more readily discoverable and useable across
augur-based tools. This commit also adds two related functions to the utils
module to enable loading auspice JSONs as Bio.Phylo instances.

Closes #250.

trvrb commented:
This is mostly okay with me in general. I was liking how self-contained and readable the different `augur/` modules were. Quite nice not to have too much redirection. But the conversion of JSON to BioPhylo seems common. Is there a better name for `all_parents` in this case? Seems esoteric?

trvrb commented on augur/utils.py:
```diff
@@ -453,3 +455,74 @@ def nthreads_value(value):
+    # Next, annotate each node with its parent.
+    for node in tree.find_clades():
+        if node == tree.root:
+            node.up = None
```
I'd think that `node.parent` would be the more usual naming, but could be wrong.

huddlej commented:
Ok, I've renamed `all_parents` to the more verbose and descriptive `get_parent_name_by_child_name_for_tree` and also renamed the parent attribute from `up` to `parent`.
